storage/jellyfish_merkle/src/lib.rs

```diff
@@ -54,3 +60,398 @@ impl From<TreeUpdateBatch> for (NodeBatch, StaleNodeIndexBatch) {
         (batch.node_batch, batch.stale_node_index_batch)
     }
 }
+
+/// The Jellyfish Merkle tree data structure. See [`crate`] for description.
+pub struct JellyfishMerkleTree<'a, R: 'a + TreeReader> {
+    reader: &'a R,
+}
+
+impl<'a, R> JellyfishMerkleTree<'a, R>
+where
+    R: 'a + TreeReader,
+{
+    /// Creates a `JellyfishMerkleTree` backed by the given [`TreeReader`].
+    pub fn new(reader: &'a R) -> Self {
+        Self { reader }
+    }
+
+    /// This is a convenient function that calls
+    /// [`put_blob_set`](JellyfishMerkleTree::put_blob_set) with a series of `keyed_blob_set`.
```

Comment:
`put_blob_sets`? Also should it say "a single `keyed_blob_set`" instead of "a series"?